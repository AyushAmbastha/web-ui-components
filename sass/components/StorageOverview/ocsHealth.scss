.ocs-health__body {
  max-height: 130px;
}

.ocs-health__icon--error {
  font-size: 1.333em;
  color: red;
  line-height: 1.333em;
}

.ocs-health__icon--ok {
  font-size: 1.333em;
  color: yellowgreen;
  line-height: 1.333em;
}

.ocs-health__icon--warning {
  font-size: 1.333em;
  color: goldenrod;
  line-height: 1.333em;
}

.ocs-ocshealth__row-status-item-text {
  margin-left: 1em;
  margin-top: 0.3em;
  font-size: 1.333em;
}

.ocs-ocshealth__row-status-item {
  display: flex;
  margin-right: 0.5em;
  flex-shrink: 0;
}
