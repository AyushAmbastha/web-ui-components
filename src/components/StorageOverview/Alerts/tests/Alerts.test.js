import React from 'react';
import { shallow, render } from 'enzyme';

import { Alerts, AlertsConnected } from '../Alerts';
import { default as AlertsFixtures } from '../fixtures/Alerts.fixture';
import { default as StorageOverviewFixtures } from '../../fixtures/StorageOverview.fixture';
import { StorageOverviewContext } from '../../StorageOverviewContext';

// eslint-disable-next-line react/prop-types
const testAlerts = ({ props }) => <Alerts {...props} />;

describe('<Alerts />', () => {
  AlertsFixtures.forEach(fixture => {
    it(`renders ${fixture.name} correctly`, () => {
      const component = shallow(testAlerts(fixture));
      expect(component).toMatchSnapshot();
    });
  });
  it('renders correctly with Provider', () => {
    const component = render(
      <StorageOverviewContext.Provider value={StorageOverviewFixtures[2].props}>
        <AlertsConnected />
      </StorageOverviewContext.Provider>
    );
    expect(component).toMatchSnapshot();
  });
});
