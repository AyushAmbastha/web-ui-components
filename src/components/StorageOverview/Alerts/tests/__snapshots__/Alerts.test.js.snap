// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Alerts /> renders Alerts correctly 1`] = `
<DashboardCard
  className={null}
>
  <DashboardCardHeader
    className={null}
  >
    <DashboardCardTitle
      className={null}
    >
      Alerts
    </DashboardCardTitle>
  </DashboardCardHeader>
  <DashboardCardBody
    LoadingComponent={[Function]}
    isLoading={false}
  >
    <AlertsBody>
      <AlertItem
        alert={
          Object {
            "annotations": Object {
              "description": "Disk inaccessible on host ip-10-0-154-62.ec2.internal (device xvda)",
              "message": "Disk is inaccessible",
              "storage_type": "ceph",
            },
            "labels": Object {
              "alertname": "CephOSDDiskUnavailable",
              "severity": "critical",
            },
          }
        }
        key="alert-0"
      />
      <AlertItem
        alert={
          Object {
            "annotations": Object {
              "description": "Data recovery has been active for over 2h. Contact Support",
              "message": "Data recovery is slow",
              "storage_type": "ceph",
            },
            "labels": Object {
              "alertname": "CephDataRecoveryTakingTooLong",
              "severity": "warning",
            },
          }
        }
        key="alert-1"
      />
    </AlertsBody>
  </DashboardCardBody>
</DashboardCard>
`;

exports[`<Alerts /> renders Data Recovery correctly 1`] = `
<DashboardCard
  className={null}
>
  <DashboardCardHeader
    className={null}
  >
    <DashboardCardTitle
      className={null}
    >
      Alerts
    </DashboardCardTitle>
  </DashboardCardHeader>
  <DashboardCardBody
    LoadingComponent={[Function]}
    isLoading={false}
  >
    <AlertsBody>
      <AlertItem
        alert={
          Object {
            "annotations": Object {
              "description": "Data recovery has been active for over 2h. Contact Support",
              "message": "Data recovery is slow",
              "storage_type": "ceph",
            },
            "labels": Object {
              "alertname": "CephDataRecoveryTakingTooLong",
              "severity": "warning",
            },
          }
        }
        key="alert-0"
      />
    </AlertsBody>
  </DashboardCardBody>
</DashboardCard>
`;

exports[`<Alerts /> renders Disk Inaccessible correctly 1`] = `
<DashboardCard
  className={null}
>
  <DashboardCardHeader
    className={null}
  >
    <DashboardCardTitle
      className={null}
    >
      Alerts
    </DashboardCardTitle>
  </DashboardCardHeader>
  <DashboardCardBody
    LoadingComponent={[Function]}
    isLoading={false}
  >
    <AlertsBody>
      <AlertItem
        alert={
          Object {
            "annotations": Object {
              "description": "Disk inaccessible on host ip-10-0-154-62.ec2.internal (device xvda)",
              "message": "Disk is inaccessible",
              "storage_type": "ceph",
            },
            "labels": Object {
              "alertname": "CephOSDDiskUnavailable",
              "severity": "critical",
            },
          }
        }
        key="alert-0"
      />
    </AlertsBody>
  </DashboardCardBody>
</DashboardCard>
`;

exports[`<Alerts /> renders Empty Alerts correctly 1`] = `""`;

exports[`<Alerts /> renders correctly with Provider 1`] = `
<div
  class="card-pf kubevirt-dashboard__card"
>
  <div
    class="card-pf-heading kubevirt-dashboard__card-heading"
  >
    <h2
      class="card-pf-title kubevirt-dashboard__card-title"
    >
      Alerts
    </h2>
  </div>
  <div
    class="card-pf-body"
  >
    <div
      class="kubevirt-alert__alerts-body"
    >
      <div
        class="kubevirt-alert__item"
      >
        <span
          aria-hidden="true"
          class="fa fa-exclamation-circle kubevirt-alert__icon kubevirt-alert__icon--critical"
        />
        <div
          class="kubevirt-alert__item-message"
        >
          Disk inaccessible on host ip-10-0-154-62.ec2.internal (device xvda)
        </div>
      </div>
      <div
        class="kubevirt-alert__item"
      >
        <span
          aria-hidden="true"
          class="fa fa-exclamation-triangle kubevirt-alert__icon kubevirt-alert__icon--warning"
        />
        <div
          class="kubevirt-alert__item-message"
        >
          Data recovery has been active for over 2h. Contact Support
        </div>
      </div>
    </div>
  </div>
</div>
`;
